tests/System.Buffers.Primitives.Tests/BasicUnitTests.cs

```diff
@@ -121,13 +121,29 @@ public void ByteReadOnlySpanCtorWithRangeValidCases(byte[] bytes, int start, int
         [Fact]
         public void ByteSpanCtorWithRangeThrowsArgumentExceptionOnNull()
         {
-            Assert.Throws<ArgumentNullException>(() => { Span<byte> span = new Span<byte>(null, 0, 0); });
+            try
```

Comment:
First, I think we should remove the tests that are duplicates of ones in CoreFx. Also, I'd move any non-duplicate tests that are specifically testing span to the CoreFx tests.

Then, I'd also look at how CoreFx uses the "DontBox" concept in the tests to effectively do exactly what this code is doing. It ends up being one line and cleaner looking with the help of a small static helper.